packages/macros/src/babel/app-ember-satisfies.ts

```diff
@@ -0,0 +1,40 @@
+import type { NodePath } from '@babel/traverse';
+import type { types as t } from '@babel/core';
+import type State from './state';
+import { satisfies } from 'semver';
+import error from './error';
+import { assertArray } from './evaluate-json';
+
+const packageName = 'ember-source';
+
+export default function appEmberSatisfies(path: NodePath<t.CallExpression>, state: State): boolean {
+  if (path.node.arguments.length !== 1) {
+    throw error(path, `dependencySatisfies takes exactly one argument, you passed ${path.node.arguments.length}`);
+  }
+  const [range] = path.node.arguments;
+  if (range.type !== 'StringLiteral') {
+    throw error(
+      assertArray(path.get('arguments'))[1],
+      `the second argument to dependencySatisfies must be a string literal`
+    );
+  }
+  try {
+    let root = state.packageCache.get(state.packageCache.appRoot);
+
+    if (!root?.hasDependency(packageName)) {
```

Comment:
I thought this was a copyPasta error 🍝 because the packageName is in module scope.. . it's only used in this try block so we could just make it a const in here 🤔

Reply:
I mean that's premature optimisation right there, v8 will 100% inline it if it's a const 🤷

Reply:
yea, i mean, the activation energy to change it is worth more than anything the computer would do.
(and what energy is spent discussing such changes).

Additionally to my prior point, I use consts like this to avoid typos in strings.
I _hate_ debugging string typos. So if I only write the word once, I can't have mismatches. :tada: